### PR TITLE
BAEL-2213 Added custom dialect for SQLite database.

### PR DESCRIPTION
**Pull request by sandy03934**

### CODE REVIEW

File: spring-data-rest/pom.xml [hunk 1]
```diff
@@ -33,6 +33,15 @@
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+        </dependency>
+        <!--<dependency>-->
```
Comment (jzheaux) [inline spring-data-rest/pom.xml]: Commenting this out confuses me. Can you explain?

File: spring-data-rest/pom.xml [hunk 2]
```diff
@@ -43,6 +52,8 @@
     </build>
 
     <properties>
+        <start-class>com.baeldung.SpringDataRestApplication</start-class>
```
Comment (jzheaux) [inline spring-data-rest/pom.xml]: Does this break other demos in the same module?

Comment (sandy03934) [reply spring-data-rest/pom.xml]: No it doesn't. The Spring-boot Maven plugin in parent-boot-2 module expects the start-class property as well. So added the value

File: spring-data-rest/pom.xml (hunk 1 above)
Comment (sandy03934) [reply spring-data-rest/pom.xml]: The project is by default working on H2 Database. So leaving this commented. We can uncomment if we want to use the SQLite database.

**Comment by sandy03934** [comment]
Added some comments to reflect the reason behind the pom changes

**Comment by jzheaux** [comment]
Closed in favor of #5822

**Pull request closed without merging**